backend/migrations/2023-07-15-152628_squashed_setup/up.sql

```diff
@@ -0,0 +1,461 @@
+    --seed_planting_depth_de text,
+    --seed_weight_1000_de text,
+    --seed_weight_1000_en text,
+    seed_weight_1000 double precision,
+    --machine_cultivation_possible boolean,
+    --edible_uses_de text,
+    CONSTRAINT plant_detail_heat_zone_check CHECK (((heat_zone IS NULL) OR ((heat_zone >= 0) AND (heat_zone <= 13)))),
+    CONSTRAINT plant_detail_preferable_permaculture_zone_check CHECK (((preferable_permaculture_zone IS NULL) OR ((preferable_permaculture_zone >= '-1'::integer) AND (preferable_permaculture_zone <= 6))))
+);
+
+CREATE SEQUENCE plant_detail_id_seq
+    AS integer
+    START WITH 1
+    INCREMENT BY 1
+    NO MINVALUE
+    NO MAXVALUE
+    CACHE 1;
+
+CREATE TABLE plantings (
+    id uuid NOT NULL,
+    layer_id integer NOT NULL,
+    plant_id integer NOT NULL,
+    x integer NOT NULL,
+    y integer NOT NULL,
+    width integer NOT NULL,
+    height integer NOT NULL,
+    rotation real NOT NULL,
+    scale_x real NOT NULL,
+    scale_y real NOT NULL,
+    add_date date,
+    remove_date date,
+    create_date date DEFAULT CURRENT_DATE NOT NULL,
+    delete_date date
+);
+
+
+
+CREATE TABLE relations (
+    plant1 integer NOT NULL,
+    plant2 integer NOT NULL,
+    relation relation_type NOT NULL,
+    note text
+);
+
+
+CREATE TABLE seeds (
+    id integer NOT NULL,
+    name text NOT NULL,
+    harvest_year smallint NOT NULL,
+    use_by date,
+    origin text,
+    taste text,
+    yield text,
+    quantity quantity NOT NULL,
+    quality quality,
+    price smallint,
+    generation smallint,
+    notes text,
+    variety text,
+    plant_id integer,
+    owner_id uuid NOT NULL
+);
+
+
+CREATE SEQUENCE seeds_id_seq
+    AS integer
+    START WITH 1
+    INCREMENT BY 1
+    NO MINVALUE
+    NO MAXVALUE
+    CACHE 1;
+
+
+ALTER TABLE ONLY layers ALTER COLUMN id SET DEFAULT nextval('layers_id_seq'::regclass);
+
+
+ALTER TABLE ONLY maps ALTER COLUMN id SET DEFAULT nextval('maps_id_seq'::regclass);
+
+
+ALTER TABLE ONLY plants ALTER COLUMN id SET DEFAULT nextval('plant_detail_id_seq'::regclass);
+
+
+ALTER TABLE ONLY seeds ALTER COLUMN id SET DEFAULT nextval('seeds_id_seq'::regclass);
+
+
+ALTER TABLE ONLY layers
+    ADD CONSTRAINT layers_pkey PRIMARY KEY (id);
+
+
+ALTER TABLE ONLY maps
+    ADD CONSTRAINT map_unique_name UNIQUE (name);
+
+
+ALTER TABLE ONLY maps
+    ADD CONSTRAINT maps_pkey PRIMARY KEY (id);
+
+
+ALTER TABLE ONLY plants
+    ADD CONSTRAINT plant_detail_pkey PRIMARY KEY (id);
+
+ALTER TABLE ONLY plants
+    ADD CONSTRAINT plant_unique_name_key UNIQUE (unique_name);
+
+
+ALTER TABLE ONLY plantings
+    ADD CONSTRAINT plantings_pkey PRIMARY KEY (id);
+
+
+ALTER TABLE ONLY relations
+    ADD CONSTRAINT relations_pkey PRIMARY KEY (plant1, plant2);
+
+
+ALTER TABLE ONLY seeds
+    ADD CONSTRAINT seeds_pkey PRIMARY KEY (id);
+
+
+CREATE TRIGGER check_layer_type_before_insert_or_update BEFORE INSERT OR UPDATE ON plantings FOR EACH ROW EXECUTE FUNCTION check_layer_type();
```

Comment:
This is a bit out-of-place here. The rest of the file is grouped differently.

Reply:
i moved it to the bottom of the file for now.